backend/src/BuiltinExecution/Libs/HttpClient.fs

```diff
@@ -429,24 +511,18 @@ let fns (config : Configuration) : List<BuiltInFn> =
 
                 return DRecord(typ, typ, [], Map fields) |> resultOk
 
-              | Error(BadUrl details) -> return resultErrorStr $"Bad URL: {details}"
-              | Error(Timeout) -> return resultErrorStr $"Request timed out"
-              | Error(NetworkError) -> return resultErrorStr $"Network error"
-              | Error(Other details) -> return resultErrorStr details
-            }
-
-          | Error reqHeadersErr, _ ->
-            uply {
-              match reqHeadersErr with
-              | BadInput details -> return resultErrorStr details
-              | TypeMismatch details ->
-                return raiseUntargetedRTE (RuntimeError.oldError details)
-            }
+              | Error(err) ->
+                let err = err |> RequestError.toDT
```

Comment:
I find the multiple errors confusing. Could you consolidate them into a single place before calling RequestError.toDT.

Reply:
These aren't all the same type, AFAICT. BadMethod is a RequestError so that's fine, but BadHeader is a separate type (it should be wrapped in a RequestHeader.BadHeader enum).

The reason there's no typeerror is because they're all converted to Dvals using toDT. This is why I'm suggesting having a `let result = ....` where the f# type of `result` is `Result<Dval, RequestError>`. Then you have the type safety of knowing all the branches have the same type in F#, and can call `toDT` at the end.